src/modules/map/components/CompareSlider.vue

```diff
@@ -1,53 +1,109 @@
 <script setup>
 import { FontAwesomeIcon } from '@fortawesome/vue-fontawesome'
-import { computed, defineProps, onMounted, ref, toRefs, watch } from 'vue'
+import { unByKey } from 'ol/Observable'
+import { computed, inject, nextTick, onMounted, ref, watch } from 'vue'
 import { useStore } from 'vuex'
 
 import { round } from '@/utils/numberUtils'
-const props = defineProps({
-    clientWidth: {
-        type: Number,
-        default: window.innerWidth,
-    },
-})
+const getMap = inject('getMap')
 
-const { clientWidth } = toRefs(props)
+const preRenderKey = ref(null)
+const postRenderKey = ref(null)
+const compareSliderOffset = ref(0)
 
 const compareRatio = ref(-0.5)
-
 const store = useStore()
 const storeCompareRatio = computed(() => store.state.ui.compareRatio)
+const clientWidth = computed(() => store.state.ui.width)
 const compareSliderPosition = computed(() => {
     return {
         left: compareRatio.value * 100 + '%',
     }
 })
+const visibleLayerOnTop = computed(() => store.getters.visibleLayerOnTop)
 
 watch(storeCompareRatio, (newValue) => {
     compareRatio.value = newValue
+    slice()
+})
+
+watch(visibleLayerOnTop, () => {
+    nextTick(slice())
 })
 
 onMounted(() => {
     compareRatio.value = storeCompareRatio.value
+    slice()
 })
 
-function grabSlider() {
+function slice() {
+    if (preRenderKey.value != null && postRenderKey.value != null) {
+        unByKey(preRenderKey.value)
+        unByKey(postRenderKey.value)
+        preRenderKey.value = null
+        postRenderKey.value = null
+    }
+    const topVisibleLayer = getMap()
+        ?.getAllLayers()
+        .find((layer) => {
+            return visibleLayerOnTop.value && layer.get('id') === visibleLayerOnTop.value.getID()
+        })
+    if (topVisibleLayer) {
+        preRenderKey.value = topVisibleLayer.on('prerender', onPreRender)
+        postRenderKey.value = topVisibleLayer.on('postrender', onPostRender)
+    }
+    getMap().render()
+}
+
+function onPreRender(event) {
+    const ctx = event.context
+    // the offset is there to ensure we get to the slider line, and not the border of the element
+    let width = ctx.canvas.width
+    if (compareRatio.value < 1.0 && compareRatio.value > 0.0) {
+        width = ctx.canvas.width * compareRatio.value + 20
+    }
+
+    ctx.save()
+    ctx.beginPath()
+    ctx.rect(0, 0, width, ctx.canvas.height)
+    ctx.clip()
+}
+
+function onPostRender(event) {
+    event.context.restore()
+    getMap().render()
```

Comment:
I'm not sure this force render is useful, as you are in a post render hook. That might re-render, and then re-trigger this hook

Reply:
I'll remove it and check if any problems arise